src/ui/widgets/ChoiceButton/choiceButton.tsx

```diff
@@ -74,61 +100,59 @@ export const ChoiceButtonComponent = (
   // Number of buttons to create
   const numButtons = options.length || 1;
   // Determine width and height of buttons if horizontal or vertically placed
-  const buttonHeight = horizontal ? height : height / numButtons - 4;
-  const buttonWidth = horizontal ? width / numButtons - 4 : width;
-
-  const style: CSSProperties = {
-    height: buttonHeight,
-    width: buttonWidth,
-    textAlignLast: "center",
-    cursor: enabled ? "default" : "not-allowed",
-    color: foregroundColor?.toString(),
-    ...font.css()
-  };
+  const buttonHeight = horizontal ? height : height / numButtons;
+  const buttonWidth = horizontal ? width / numButtons : width;
 
-  function handleClick(index: number) {
+  const handleChange = (
+    event: React.MouseEvent<HTMLElement>,
+    newSelect: number
+  ) => {
     // Write to PV
     if (pvName) {
-      writePv(pvName, new DType({ doubleValue: index }));
+      writePv(pvName, new DType({ doubleValue: newSelect }));
     }
-  }
-
-  // Iterate over items to create buttons
-  const elements: Array<JSX.Element> = [];
-  options.forEach((item: string | null | undefined, idx: number) => {
-    if (typeof item === "string") {
-      elements.push(
-        <button
-          className={classes.ChoiceButton}
-          disabled={enabled ? false : true}
-          onClick={() => handleClick(idx)}
-          style={{
-            ...style,
-            backgroundColor:
-              selected === idx
-                ? selectedColor.toString()
-                : backgroundColor.toString(),
-            boxShadow:
-              selected === idx
-                ? `inset 0px ${Math.round(height / 6)}px ${Math.round(
-                    height / 4
-                  )}px 0px rgba(0,0,0,0.3)`
-                : "none"
-          }}
-          key={item}
-        >
-          {item}
-        </button>
-      );
-    }
-  });
+    setSelected(newSelect);
+  };
 
   return (
-    <div
-      style={{ display: "flex", flexDirection: horizontal ? "row" : "column" }}
+    <ToggleButtonGroup
+      exclusive
+      fullWidth={true}
+      disabled={!enabled}
+      value={selected}
+      onChange={handleChange}
+      orientation={horizontal ? "horizontal" : "vertical"}
+      sx={{
+        display: "flex",
+        height: "100%",
+        width: "100%"
+      }}
     >
-      {elements}
-    </div>
+      {options
+        .filter(item => typeof item === "string")
+        .map((item, idx) => (
+          <ToggleButton
+            key={item}
+            value={idx}
+            sx={{
+              maxWidth: buttonWidth,
+              maxHeight: buttonHeight,
+              fontFamily: font,
+              color: foregroundColor.toString(),
+              backgroundColor: backgroundColor.toString(),
+              "&.Mui-selected": {
+                backgroundColor: selectedColor.toString()
+              },
+              "&.Mui-selected:hover, &:hover": {
```

Comment:
I think the hover needs to be split out here, as when I hover over an unselected component it changes to the selected colour. So:

- On select and hover - change selected color and opacity
- On hover - change opacity